File: src/main/java/br/com/correios/enderecador/util/GerenciadorPropriedades.java
package br.com.correios.enderecador.util;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class GerenciadorPropriedades {
    private final Properties p = new Properties();

    private final String arquivoPropriedade;

    public GerenciadorPropriedades(String arquivo) {
        this.arquivoPropriedade = arquivo;
        loadProperties();
    }

    public String getProperties(String v) {
        return this.p.getProperty(v);
    }

    public void setProperties(String p, String v) {
        this.p.setProperty(p, v);
    }

    private void loadProperties() {
        InputStream in = getClass().getResourceAsStream(this.arquivoPropriedade);
        if (in != null)
            try {
                this.p.load(in);
            } catch (IOException ex) {
                ex.printStackTrace();
            } finally {
                try {
                    in.close();
                } catch (IOException ex) {
                    ex.printStackTrace();
                }
            }
    }
}

File: src/main/kotlin/br/com/correios/enderecador/util/GerenciadorPropriedades.kt
package br.com.correios.enderecador.util

import java.util.Properties
import java.io.IOException

class GerenciadorPropriedades(private val arquivoPropriedade: String) {
    private val p = Properties()

    init {
        loadProperties()
    }

    fun getProperties(v: String?): String {
        return p.getProperty(v)
    }

    fun setProperties(p: String?, v: String?) {
        this.p.setProperty(p, v)
    }

    private fun loadProperties() {
        val `in` = javaClass.getResourceAsStream(arquivoPropriedade)
        if (`in` != null) try {
            p.load(`in`)
        } catch (ex: IOException) {
            ex.printStackTrace()
        } finally {
            try {
                `in`.close()
            } catch (ex: IOException) {
                ex.printStackTrace()
            }
        }
    }
}
